Firestore/core/src/firebase/firestore/remote/connectivity_monitor_apple.mm

```diff
@@ -107,9 +111,31 @@ explicit ConnectivityMonitorApple(
       LOG_DEBUG("Couldn't set reachability queue");
       return;
     }
+
+#if TARGET_OS_IOS || TARGET_OS_TV
+    this->observer_ = [[NSNotificationCenter defaultCenter]
+        addObserverForName:UIApplicationWillEnterForegroundNotification
+                    object:nil
+                     queue:[NSOperationQueue mainQueue]
+                usingBlock:^(NSNotification* note) {
+                  SCNetworkReachabilityFlags flags{};
+                  if (SCNetworkReachabilityGetFlags(reachability_, &flags)) {
+                    auto status = ToNetworkStatus(flags);
+                    if (status != NetworkStatus::Unavailable) {
+                      this->InvokeCallbacks(status);
```

Comment:
Internal callbacks need to run on the `AsyncQueue`. You might want to factor out an `OnEnteredForeground` member (and put it near `OnReachabilityChanged` to make this parallel clearer. Something like this:

```c++
void OnEnteredForeground() {
  SCNetworkReachabilityFlags flags{};
  if (!SCNetworkReachabilityGetFlags(reachability_, &flags)) return;

  queue()->Enqueue([this, flags] {
    auto status = ToNetworkStatus(flags);
    if (status != NetworkStatus::Unavailable) {
      this->InvokeCallbacks(status);
    } else {
      this->MaybeInvokeCallbacks(status);
    }
  });
}
```

Reply:
Done.